worlds/blasphemous/Rules.py

```diff
@@ -1,4458 +1,1373 @@
+    
+    def verdiales(self, state: CollectionState) -> bool:
+        return state.has("Verdiales of the Forsaken Hamlet", self.player)
+    
+    def cante(self, state: CollectionState) -> bool:
+        return state.has("Cante Jondo of the Three Sisters", self.player)
+    
+    def cantina(self, state: CollectionState) -> bool:
+        return state.has("Cantina of the Blue Rose", self.player)
 
-def can_break_holes(state: CollectionState, player: int) -> bool:
-    return (
-        state.has_any({"Charged Skill", "Dive Skill"}, player) 
-        or (
-            state.has("Lunge Skill", player, 3)
-            and state.has("Dash Ability", player)
+    def aubade(self, state: CollectionState) -> bool:
+        return (
+            state.has("Aubade of the Nameless Guardian", self.player)
+            and self.total_fervour(state) >= 90
+        )
+    
+    def tirana(self, state: CollectionState) -> bool:
+        return (
+            state.has("Tirana of the Celestial Bastion", self.player)
+            and self.total_fervour(state) >= 90
         )
-        or state.has_group("prayer", player)
-        or aubade(state, player)
-        or tirana(state, player)
-    )
-
-
-def can_break_tirana(state: CollectionState, logic: int, player: int) -> bool:
-    return tirana(state, player) if logic >= 2 else False
-
-
-def can_dive_laser(state: CollectionState, logic: int, player: int) -> bool:
-    return state.has("Dive Skill", player, 3) if logic >= 2 else False
-
-
-def can_walk_on_root(state: CollectionState, player: int) -> bool:
-    return state.has("Three Gnarled Tongues", player)
-
-
-def can_climb_on_root(state: CollectionState, player: int) -> bool:
-    return state.has_all({"Three Gnarled Tongues", "Wall Climb Ability"}, player)
-
-
-def can_survive_poison(state: CollectionState, logic: int, player: int, number: int) -> bool:
-    if number == 1:
-        if logic >= 2:
-            return True
-        elif logic == 1:
-            return state.has_any({"Silvered Lung of Dolphos", "Tiento to your Thorned Hairs"}, player)
-        elif logic == 0:
-            return state.has("Silvered Lung of Dolphos", player)
-    elif number == 2:
-        if logic >= 1:
-            return state.has_any({"Silvered Lung of Dolphos", "Tiento to your Thorned Hairs"}, player)
-        else:
-            return state.has("Silvered Lung of Dolphos", player)
-    elif number == 3:
-        if logic >= 2 and total_fervour(state, player) >= 120:
-            return state.has_any({"Silvered Lung of Dolphos", "Tiento to your Thorned Hairs"}, player)
-        else:
-            return state.has("Silvered Lung of Dolphos", player)
 
+    def ruby(self, state: CollectionState) -> bool:
+        return state.has("Cloistered Ruby", self.player)
+    
+    def tiento(self, state: CollectionState) -> bool:
+        return state.has("Tiento to my Sister", self.player)
+    
+    def any_small_prayer(self, state: CollectionState) -> bool:
```

Comment:
Isn't this missing Campanillero, Mirabras, Romance, Saeta, Seguiriya, Verdiales, and Zambra?